core/server/data/migrations/versions/5.0/2022-05-04-15-24-map-existing-emails-to-default-newsletter.js

```diff
@@ -0,0 +1,42 @@
+const logging = require('@tryghost/logging');
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Linking existing emails and related posts to the default newsletter');
+
+        // Get the default newsletter
+        // Note we intentionally use the default newsletter slug instead of the usual orderBy logic
+        let newsletter = await knex('newsletters')
+            .where('slug', 'default-newsletter')
+            .first('id');
+
+        if (!newsletter) {
+            // Fall back to orderBy - just in case
+            logging.warn(`Original default newsletter not found - using first in sort order`);
+            newsletter = await knex('newsletters')
+                .where('status', 'active')
+                .orderBy('sort_order', 'asc')
+                .orderBy('created_at', 'asc')
+                .orderBy('id', 'asc')
+                .first('id');
+        }
+
+        if (!newsletter) {
+            logging.error(`Newsletter not found - skipping`);
+            return;
+        }
+
+        // Set newsletter ID only on posts with related email records without a newsletter assigned
```

Comment:
Can we add a log line here to print out the newsletter ID we've identified and will use in the queries? Helps with debugging 😛